app/src/main/java/org/dicekeys/app/openpgp/Packet.kt

```diff
@@ -0,0 +1,28 @@
+package org.dicekeys.app.openpgp
+
+import com.google.common.io.ByteStreams
+import java.security.MessageDigest
+
+abstract class Packet {
+    abstract val ctb : Int
+    abstract val body : ByteArray
+
+    abstract fun hash(digest: MessageDigest)
+
+    fun toByteArray(sizeInBytes: Int = 1): ByteArray {
+        val out = ByteStreams.newDataOutput()
+        out.writeByte(ctb)
+        // hardcoded byte ? it should be variable, but it's ok for our usecase
+        out.writeByte(body.size)
+        out.write(body)
+        return out.toByteArray()
+    }
+
+    companion object{
```

Comment:
Looks like a clean  way to manage the constants.  Can you include comments that point to the specs where these constants come from?

Reply:
Done